File: SourceCodes/C/recursion.c
#include <stdio.h>

int fib(int val) {
    if (val == 1 || val == 0) {
        return 1;
    }
    return fib(val - 1) + fib(val - 2);
}

int main(int argc, char* argv[]) {
    printf("%i\n",fib(44));
    return 0;
}

File: SourceCodes/Rust/recursion.rs
fn main() {
    println!("{}",fib(44));
}

fn fib(val: i128) -> i128 {
    if val == 1 || val == 0 {
        return 1;
    }
    return fib(val - 1) + fib(val - 2);
}